pkg/localstore/subscription_push.go

```diff
@@ -105,6 +120,23 @@ func (db *DB) SubscribePush(ctx context.Context) (c <-chan swarm.Chunk, stop fun
 					db.logger.Debugf("localstore push subscription iteration: %v", err)
 					return
 				}
+
+				// save last Item from this iteration in order to know were
+				// to stop on next iteration
+				if lastItem != nil && toItemKey == nil {
+					var lastItemKey []byte
+					// move 'toItemKey' to point to last item in previous iteration
+					lastItemKey, err = db.pushIndex.ItemKey(*lastItem)
+					if err != nil {
+						return
+					}
+
+					toItemKey = append(lastItemKey[:0:0], lastItemKey...)
```

Comment:
the three index slice is rather exotic, why no just `toItemKey = lastItemKey`? `lastItemKey` is scoped in this block. I don't see why you need this overhead?
you could also just `toItemKey, err = db.pushIndex.ItemKey(*lastItem)` directly